File: scripts/03-test_data.R
#### Preamble ####
# Purpose: Tests... [...UPDATE THIS...]
# Author: Rohan Alexander [...UPDATE THIS...]
# Date: 11 February 2023 [...UPDATE THIS...]
# Contact: rohan.alexander@utoronto.ca [...UPDATE THIS...]
# License: MIT
# Pre-requisites: [...UPDATE THIS...]
# Any other information needed? [...UPDATE THIS...]


#### Workspace setup ####
library(tidyverse)
# [...UPDATE THIS...]

#### Test data ####

# Test 1: Check for "n/a" Replacement
test_na_replacement <- function(cleaned_avg_age_death_data) {
  # Check if any "n/a" values remain in the data
  has_na <- any(cleaned_avg_age_death_data == "n/a")
  
  if (!has_na) {
    print("Test 1 Passed: All 'n/a' values have been replaced with 0.")
  } else {
    print("Test 1 Failed: Some 'n/a' values remain in the data.")
  }
}

# Run Test 1
test_na_replacement(cleaned_avg_age_death_data)
test_na_replacement(cleaned_death_data)


# Test2: Check for Negative Numbers
test_negative_numbers <- function(cleaned_avg_age_death_data) {
  # Define the relevant columns to check for negative numbers
  numeric_columns <- c(
    "Total decedents",
    "Average age of death, all decedents",
    "Male decedents",
    "Average age of death, male decedents",
    "Female decedents",
    "Average age of death, female decedents",
    "Transgender/Non-binary/Two-Spirit decedents",
    "Avg age of death, transgender/non-binary/two-spirit decedents"
  )
  
  # Check for negative values in the specified columns
  has_negative <- any(sapply(cleaned_avg_age_death_data[numeric_columns], function(x) any(x < 0)))
  
  if (!has_negative) {
    print("Negative Number Test Passed: No negative values found in the relevant columns.")
  } else {
    print("Negative Number Test Failed: Negative values found in the relevant columns.")
  }
}

# Run the Negative Number Test
test_negative_numbers(cleaned_avg_age_death_data)


# Test2: Check for Negative Numbers
test_negative_numbers <- function(cleaned_death_data) {
  # Define the relevant columns to check for negative numbers
  numeric_columns <- c(
    "Total decedents",
    "Male",
    "Female",
    "Transgender/Non-binary/Two-Spirit"
  )
  
  # Check for negative values in the specified columns
  has_negative <- any(sapply(cleaned_death_data[numeric_columns], function(x) any(x < 0)))
  
  if (!has_negative) {
    print("Negative Number Test Passed: No negative values found in the relevant columns.")
  } else {
    print("Negative Number Test Failed: Negative values found in the relevant columns.")
  }
}

# Run the Negative Number Test
test_negative_numbers(cleaned_death_data)

File: scripts/03-test_data.py
#### Preamble ####
# Purpose: Tests for "n/a" Replacement and Negative Numbers.
# Author: Vivian Guo
# Date: 23 September 2024
# Contact: vivian.guo@mail.utoronto.ca
# License: N/A
# Pre-requisites: N/A

#### Workspace setup ####
import pandas as pd

#### Test data ####

# Test 1: Check for "n/a" Replacement
def test_na_replacement(df, df_name):
    # Check if any 'n/a' strings remain in the data
    has_na = (df == "n/a").any().any()

    if not has_na:
        print(f"Test 1 Passed: All 'n/a' values have been replaced with 0 in {df_name}.")
    else:
        print(f"Test 1 Failed: Some 'n/a' values remain in the data in {df_name}.")

# Run Test 1
test_na_replacement(cleaned_avg_age_death_data, "cleaned_avg_age_death_data")
test_na_replacement(cleaned_death_data, "cleaned_death_data")

# Test 2: Check for Negative Numbers in cleaned_avg_age_death_data
def test_negative_numbers_avg_age(df):
    numeric_columns = [
        "total_decedents",
        "Average_age_of_death_all_decedents",
        "Male_decedents",
        "Average_age_of_death_male_decedents",
        "Female_decedents",
        "Average_age_of_death_female_decedents",
        "Transgender_Non_binary_Two_Spirit_decedents",
        "Avg_age_of_death_transgender_non_binary_two_spirit_decedents"
    ]

    has_negative = (df[numeric_columns] < 0).any().any()

    if not has_negative:
        print("Negative Number Test Passed: No negative values found in the relevant columns (avg age dataset).")
    else:
        print("Negative Number Test Failed: Negative values found in the relevant columns (avg age dataset).")

# Run Test 2 on avg age dataset
test_negative_numbers_avg_age(cleaned_avg_age_death_data)

# Test 2: Check for Negative Numbers in cleaned_death_data
def test_negative_numbers_deaths(df):
    numeric_columns = [
        "total_decedents",
        "male",
        "female",
        "transgender_nonbinary_twospirit"
    ]

    has_negative = (df[numeric_columns] < 0).any().any()

    if not has_negative:
        print("Negative Number Test Passed: No negative values found in the relevant columns (death count dataset).")
    else:
        print("Negative Number Test Failed: Negative values found in the relevant columns (death count dataset).")

# Run Test 2 on death count dataset
test_negative_numbers_deaths(cleaned_death_data)
